modin/data_management/functions/groupby_function.py

```diff
@@ -115,19 +115,9 @@ def compute_reduce(df):
                 except ValueError:
                     return compute_reduce(df.copy())
 
-            if axis == 0:
-                new_columns = qc.columns
-                new_index = None
-            else:
-                new_index = query_compiler.index
-                new_columns = None
+            # TODO: try to precompute `new_index` and `new_columns`
             new_modin_frame = qc._modin_frame.groupby_reduce(
-                axis,
-                by._modin_frame,
-                _map,
-                _reduce,
-                new_columns=new_columns,
-                new_index=new_index,
+                axis, by._modin_frame, _map, _reduce
```

Comment:
Can we set `new_columns` and `new_index` based on some condition?

Reply:
Since we don't know what passed function would do with our DataFrame I'm afraid not. Let's remember our approach from #1645 where applied function transforms and changes shape of DataFrames in partitions. If we allow to apply functions like this, it's impossible to predict new indices/columns before computing.